### PR TITLE
Fix buffer overrun when parsing buffers with no trailing zero

### PR DESCRIPTION
**Pull request by JCash**
## PR checklist (remove before merging)

* [ ] Prepared the PR for release notes
	* [ ] Proper description of feature or fix (see example x?)
	* [ ] Added a #fixes ### if it fixes an issue
	* [ ] Assigned issue to a project
* [ ] Added documentation for public API changes
* [-] Is this a new feature?
	* [x] Added engine and/or editor unit tests
	* [x] Added or updated manual entry

### CODE REVIEW

File: engine/script/src/luacjson/lua_cjson.c [hunk 1]
```diff
@@ -1251,6 +1252,13 @@ static void json_next_token(json_parse_t *json, json_token_t *token)
     const json_token_type_t *ch2token = json->cfg->ch2token;
     int ch;
 
+    // DEFOLD
+    if (json->ptr >= json->data_end) {
+        token->type = T_END;
+        return;
+    }
+    // END DEFOLD
```
Comment (JCash) [inline engine/script/src/luacjson/lua_cjson.c:1260]: I'm going to try to make at least one PR back to their repository to fix this

File: engine/script/src/test/test_script_json.cpp [hunk 2]
```diff
@@ -98,6 +100,25 @@ TEST_F(ScriptJsonTest, TestJson)
...
+TEST_F(ScriptJsonTest, TestJsonToLua)
+{
+    int top = lua_gettop(L);
+
+    {
+        const char* json_original = "{\"foo\":\"bar\",\"num\":16}hello";
+        size_t json_length = 22;
+        // Make it fully dynamic so that ASAN can catch it
+        const char* json = (const char*)malloc(json_length);
+        memcpy((void*)json, (void*)json_original, json_length);
```
Comment (JCash) [inline engine/script/src/test/test_script_json.cpp:112]: This allowed me to catch it with ASAN

Comment (AGulev) [reply engine/script/src/test/test_script_json.cpp:112]: smart!

**Review by AGulev — APPROVED** [review]
👍

**Pull request merged**